### PR TITLE
FIDO metadata

### LINKED ISSUE #251: FIDO authenticator metadata
- Use the data from [FIDO Alliance Metadata Service](https://fidoalliance.org/metadata/) to get extra information about webauthn devices.
- Use the metadata `description` field for naming the registered device.

### PR DESCRIPTION
**Pull request by byewokko**
closes #251 

- Registered WebAuthn devices now have a human-readable name (if they are found in the FIDO Metadata Service).
- Preferred attestation changed to `direct`, to request extra weabuthn registration metadata from browsers.

### CODE REVIEW

File: seacatauth/authn/webauthn/service.py [hunk 1]
```diff
@@ -42,20 +48,69 @@ def __init__(self, app, service_name="seacatauth.WebAuthnService"):
...
 
+	async def initialize(self, app):
+		await self._load_fido_metadata()
+
+
 	async def _on_housekeeping(self, event_name):
 		await self._delete_expired_challenges()
 
 
+	async def _load_fido_metadata(self, speculative=True):
```
Comment (byewokko) [inline seacatauth/authn/webauthn/service.py:75]: - save the metadata in mongo collection
- load on startup if collection empty
- refresh on housekeeping

- alternatively save the JWT in /tmp and store the ETag

Comment (byewokko) [reply seacatauth/authn/webauthn/service.py:75]: there is no ETag available

File: seacatauth/authn/webauthn/service.py [hunk 2]
```diff
@@ -42,20 +57,80 @@ def __init__(self, app, service_name="seacatauth.WebAuthnService"):
...
+		"""
+		Download and decode FIDO metadata from FIDO Alliance Metadata Service (MDS) and prepare a lookup dictionary.
+		"""
+		if speculative:
```
Comment (ateska) [inline seacatauth/authn/webauthn/service.py:88]: `speculative` -> `only_if_empty`

File: seacatauth/authn/webauthn/service.py [hunk 3]
```diff
@@ -42,20 +57,80 @@ def __init__(self, app, service_name="seacatauth.WebAuthnService"):
...
+			if "aaguid" not in entry:
+				continue
+			aaguid = bytes.fromhex(entry["aaguid"].replace("-", ""))
+			metadata = entry.get("metadataStatement")
+			metadata["_id"] = aaguid
+			bulk.append(pymongo.InsertOne(metadata))
+
+		coll = await self.StorageService.collection(self.FidoMetadataServiceCollection)
+		await coll.drop()
+		result = await coll.bulk_write(bulk)
```
Comment (ateska) [inline seacatauth/authn/webauthn/service.py:130]: This should be done atomically.

**Review by ateska — CHANGES_REQUESTED** [review]

**Pull request merged**